src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1048,6 +1057,199 @@ bool needsReporting()
+            var intType = this.DeclaringCompilation.GetSpecialType(SpecialType.System_Int32);
+            var intPlaceholder = new BoundInterpolatedStringArgumentPlaceholder(arguments.AttributeSyntaxOpt!, intType, hasErrors: intType.IsErrorType());
+            // literalLength
+            analyzedArguments.Arguments.Add(intPlaceholder);
+            analyzedArguments.RefKinds.Add(RefKind.None);
+            // formattedCount
+            analyzedArguments.Arguments.Add(intPlaceholder);
+            analyzedArguments.RefKinds.Add(RefKind.None);
+
+            foreach (var parameter in parameters)
+            {
+                var argumentType = parameter?.Type ?? ContainingType;
+                analyzedArguments.Arguments.Add(new BoundInterpolatedStringArgumentPlaceholder(arguments.AttributeSyntaxOpt!, argumentType, hasErrors: argumentType.IsErrorType()));
+                analyzedArguments.RefKinds.Add(parameter?.RefKind ?? RefKind.None);
+            }
+
+            // First try without an out parameter
+            bool success = binder.TryPerformConstructorOverloadResolution(
+                handlerType,
+                analyzedArguments,
+                errorName: handlerType.Name,
+                errorLocation,
+                suppressResultDiagnostics: true,
+                BindingDiagnosticBag.Discarded,
+                memberResolutionResult: out _,
+                candidateConstructors: out _,
+                allowProtectedConstructorsOfBaseType: false);
+
+            if (!success)
+            {
+                // Retry with an out parameter
+                var boolType = this.DeclaringCompilation.GetSpecialType(SpecialType.System_Boolean);
+                analyzedArguments.Arguments.Add(new BoundInterpolatedStringArgumentPlaceholder(arguments.AttributeSyntaxOpt!, boolType, hasErrors: boolType.IsErrorType()));
+                analyzedArguments.RefKinds.Add(RefKind.Out);
+
+                success = binder.TryPerformConstructorOverloadResolution(
+                    handlerType,
+                    analyzedArguments,
+                    errorName: handlerType.Name,
+                    errorLocation,
+                    suppressResultDiagnostics: true,
+                    BindingDiagnosticBag.Discarded,
+                    memberResolutionResult: out _,
+                    candidateConstructors: out _,
+                    allowProtectedConstructorsOfBaseType: false);
+            }
+
+            if (success)
+            {
+                arguments.GetOrCreateData<ParameterWellKnownAttributeData>().InterpolatedStringHandlerArguments = parameterOrdinals;
+            }
+            else
+            {
+                // Could not find a constructor on '{0}' that matches the parameters specified in the InterpolatedStringHandlerArgumentAttribute.
+                diagnostics.Add(ErrorCode.ERR_CouldNotFindApplicableInterpolatedStringHandlerConstructor, errorLocation, handlerType);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_CouldNotFindApplicableInterpolatedStringHandlerConstructor, errorLocation, handlerType); [](http://example.com/codeflow?start=16&length=118)

I think we should worry whether there are things that are specific to this particular location that could cause a failure that might not occur at a different location (missing references, language version, etc.)